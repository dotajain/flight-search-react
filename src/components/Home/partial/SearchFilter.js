import React, { Component } from 'react';
import PropTypes from 'prop-types';
import InputRange from 'react-input-range';
import { connect } from 'react-redux';

import { handleFilter } from '../../../actions';

class SearchFilter extends Component {
  constructor(props) {
    super(props);

    this.state = {
      value: { min: 0, max: 10000 },
    };
  }

  componentWillMount() {
    this.props.handleFilter(this.state.value);
  }

  onChange = value => {
    this.setState({ value }, () => {
      this.props.handleFilter(this.state.value);
    });
  };

  render() {
    return (
      <div className="box">
        <div className="box__card">
          <h2>Refine Flight Search</h2>
          <div className="range__slider">
            <InputRange
              step={500}
              maxValue={10000}
              minValue={0}
              value={this.state.value}
              onChange={this.onChange}
            />
          </div>
        </div>
      </div>
    );
  }
}
SearchFilter.propTypes = {
  handleFilter: PropTypes.func,
};

// export default SearchFilter;

const mapStateToProps = state => ({ state });
export default connect(mapStateToProps, { handleFilter })(SearchFilter);
