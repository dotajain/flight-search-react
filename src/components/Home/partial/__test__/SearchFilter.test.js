import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../../../store';

import SearchFilter from '../SearchFilter';

global.fetch = jest.fn(
  () =>
    new Promise(resolve => {
      process.nextTick(() => resolve({ json: () => ({}) }));
    }),
);

it('Properly rendered SearchFilter Component', () => {
  const component = renderer.create(<Provider store={store}><SearchFilter /></Provider>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
