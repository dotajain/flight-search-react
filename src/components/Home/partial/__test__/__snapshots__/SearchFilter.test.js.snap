// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Properly rendered SearchFilter Component 1`] = `
<div
  className="box"
>
  <div
    className="box__card"
  >
    <h2>
      Refine Flight Search
    </h2>
    <div
      className="range__slider"
    >
      <div
        aria-disabled={false}
        className="input-range"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      >
        <span
          className="input-range__label input-range__label--min"
        >
          <span
            className="input-range__label-container"
          >
            0
          </span>
        </span>
        <div
          className="input-range__track input-range__track--background"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="input-range__track input-range__track--active"
            style={
              Object {
                "left": "0%",
                "width": "100%",
              }
            }
          />
          <span
            className="input-range__slider-container"
            style={
              Object {
                "left": "0%",
                "position": "absolute",
              }
            }
          >
            <span
              className="input-range__label input-range__label--value"
            >
              <span
                className="input-range__label-container"
              >
                0
              </span>
            </span>
            <div
              aria-controls={undefined}
              aria-labelledby={undefined}
              aria-valuemax={10000}
              aria-valuemin={0}
              aria-valuenow={0}
              className="input-range__slider"
              draggable="false"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchStart={[Function]}
              role="slider"
              tabIndex="0"
            />
          </span>
          <span
            className="input-range__slider-container"
            style={
              Object {
                "left": "100%",
                "position": "absolute",
              }
            }
          >
            <span
              className="input-range__label input-range__label--value"
            >
              <span
                className="input-range__label-container"
              >
                10000
              </span>
            </span>
            <div
              aria-controls={undefined}
              aria-labelledby={undefined}
              aria-valuemax={10000}
              aria-valuemin={0}
              aria-valuenow={10000}
              className="input-range__slider"
              draggable="false"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchStart={[Function]}
              role="slider"
              tabIndex="0"
            />
          </span>
        </div>
        <span
          className="input-range__label input-range__label--max"
        >
          <span
            className="input-range__label-container"
          >
            10000
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
